File: Ejercicios Propuestos 1/Ejercicio 16/EJERCICIO_16.cpp
#include <iostream>

using namespace std;

int main() {
    int numero;
    int suma_divisores = 0;

    cout << "Ingrese un número entero positivo: ";
    cin >> numero;

    if (numero <= 0) {
        cout << "Error: El número debe ser positivo." << endl;
    } else {
        for (int i = 1; i < numero; i++) {
            if (numero % i == 0) {
                suma_divisores += i;
            }
        }

        if (suma_divisores == numero) {
            cout << "El número " << numero << " SÍ es un número perfecto." << endl;
        } else {
            cout << "El número " << numero << " NO es un número perfecto." << endl;
        }
    }

    return 0;
}

File: Ejercicios Propuestos 1/Ejercicio 16/Ejercicio 16.py
NAME = "Renzo Leonardo Cienfuegos Cardenas"
GROUP = 2

# Un numero natural se domina perfecto cuando es igual a la suma de todos sus divisores, primos o no, excepto el mismo. por ejemplo; 28=1+2+4+7+14 escribir si un numero natural positivo es perfecto sí o no?

# Solicitamos un número al usuario
num = int(input("Número: "))

# Inicializamos la variable acumuladora
suma = 0

# Buscamos todos los divisores propios del número
for i in range(1, num):

    # Verificamos si i es divisor de num
    if num % i == 0:

        # Acumulamos el divisor encontrado
        suma += i

# Comparamos la suma de divisores con el número original
if suma == num:
    print("Es perfecto")
else:
    print("No es perfecto")